src/lib.rs

```diff
@@ -423,6 +433,32 @@ pub trait ReaderExt<'a> {
     fn read_ipv6addr(&mut self) -> Result<Ipv6Addr, Error> {
         self.read_u128be().map(Ipv6Addr::from)
     }
+
+    /// Reads null terminated string.
+    ///
+    /// Length is the maximum number of bytes to read. Reading is terminated
+    /// either when null byte is encountered or maximum number of bytes read is
+    /// reached.
+    ///
+    /// Returns Ok(v) where v is a [CString], or Err(Error::EndOfInput) if the
+    /// Reader encountered an end of the input while reading, or
+    /// Err(Error::ParseError) if null termination couldn't be found.
+    #[inline]
+    #[cfg(feature = "use_std")]
+    fn read_cstring(&mut self, max_num_bytes: usize) -> Result<CString, Error> {
+        let reader = |input: &mut Reader<'_>| -> Result<(), Error> {
+            for _ in 0..max_num_bytes {
+                if input.peek(0) {
+                    return Ok(());
```

Comment:
This leaves the NUL byte to `input` which probably is not what the user expects.

Reply:
Fixed, thanks!